docs/proposals/configuration.md

```diff
@@ -0,0 +1,481 @@
+---
+title: Configuration Files Design
+
+authors:
+  - "@kadisi"
+  - "@fisherxu"
+  
+approvers:
+  - "@kevin-wangzefeng"
+  - "@sids-b"
+
+creation-date: 2019-10-02
+
+---
+
+* [Configuration Files Design](#configuration-files-design)
+   * [Abstract](#abstract)
+   * [Goals](#goals)
+   * [Proposal](#proposal)
+      * [common apis](#common-apis)
+         * [common config struct](#common-config-struct)
+      * [cloudcore apis](#cloudcore-apis)
+         * [config struct](#config-struct)
+         * [default config file](#default-config-file)
+      * [edgecore apis](#edgecore-apis)
+         * [config struct](#config-struct-1)
+         * [default config file](#default-config-file-1)
+      * [edgesite apis](#edgesite-apis)
+         * [config struct](#config-struct-2)
+         * [default config file](#default-config-file-2)
+
+# Configuration Files Design
+
+
+## Abstract  
+
+Currently, cloudcore/edgecore/edgesite\`s configuration files are in the conf directory at the same level and have 3 configuration files, it is difficult to maintain and extend.
+
+Currently KubeEdge uses beehive's config package to analyse configuration files, but it has many drawbacks,when the program is running, it will output a lot of logs first.
+
+Is also no need to read configuration variables in multiple ways, such as commandline and env.
+
+## Goals
+
+Binary default reads yaml files in `/etc/kubeedge/config` directory and then performs the relevant logical processing.
+
+When binary run, it will first load config yaml file by `sigs.k8s.io/yaml` package rather than `beehive` package,
+
+and it also need to easily generate the configuration files for `keadm`.
+
+
+## Proposal
+
+### common apis
+
+#### common config struct
+
+defined in `pkg/common/apis/config` dir
+
+```go
+
+package config
+
+type Modules struct {
+	//default devicecontroller, edgecontroller, cloudhub
+	Enabled []string `json:"enabled,omitempty"`
+}
+
+
+```
+
+### cloudcore apis
+
+#### config struct 
+
+defined in `pkg/cloudcore/apis/config/types.go` dir
+
+```go
+
+import commonconfig "github.com/kubeedge/kubeedge/pkg/common/apis/config
+
+type CloudCoreConfig struct {
+	Kube             *KubeConfig             `json:"kube,omitempty"`
```

Comment:
A good field description comment should include:
1. Explanation of the field, what is it for
2. Default value if there is
3. Required (or Optional)
4. Additional marks e.g. deprecated.

Reply:
done